src/parser.c

```diff
@@ -936,7 +936,7 @@ static AST_HTML_ELEMENT_NODE_T* parser_parse_html_regular_element(
     close_tag = parser_parse_html_close_tag(parser);
   }
 
-  bool matches_stack = parser_check_matching_tag(parser, close_tag->tag_name->value);
+  bool matches_stack = parser_check_matching_tag(parser, hb_string_from_c_string(close_tag->tag_name->value));
```

Comment:
For the future: I feel like it would be great if we could move the `hb_string_from_c_string` to the token-level, so that `value` always returns a `hb_string_T`.